core/client/README.md

```diff
@@ -0,0 +1,364 @@
+
+In our example app, we can define the following closure to use after every
+transaction we broadcast:
+
+```go
+// After broadcast, we get a transaction hash that uniquely identifies the
+// transaction. Use the TxQuery method to verify the execution succeeded.
+checkTx := func(txHash []byte, desc string) {
+	res, err := cl.TxQuery(ctx, txHash)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if res.TxResult.Code == transactions.CodeOk.Uint32() {
+		fmt.Printf("Success: %q in transaction %x\n", desc, txHash)
+	} else {
+		log.Fatalf("Fail: %q in transaction %x, Result code %d, log: %q",
+			desc, txHash, res.TxResult.Code, res.TxResult.Log)
+	}
+}
+
+txHash, err := cl.DeployDatabase(ctx, schema, txOpts...)
+if err != nil {
+	log.Fatal(err)
+}
+checkTx(txHash, "deploy database")
+```
+
+Combined the `WithSyncBroadcast` option in `txOpts`, this use of `TxQuery` will
+ensure the transaction executed without error, otherwise the application will exit.
+
+### Dropping a Database
+
+With a successfully deployed database, use the `DropDatabase` method to delete a database:
+
+```go
+txHash, err = cl.DropDatabase(ctx, dbName, txOpts...)
+if err != nil {
+	log.Fatal(err)
+}
+checkTx(txHash, "drop database")
+```
+
+NOTE: This is only permitted if you are the `owner` of the database i.e. you deployed it.
+
+### Action Execution
+
+As with the database deploy and drop methods, action *execution* requires a
+transaction since it is used to modify data in a schema.  Use the
+`ExecuteAction` method.
+
+In the schema deployed by our example app, the action called `"tag"` will insert data:
+
+```go
+const actionName = "tag"
+args := [][]any{{"jon was here"}} // one execution, one argument
+txHash, err := cl.ExecuteAction(ctx, dbid, actionName, args, txOpts...)
+if err != nil {
+	log.Fatal(err)
+}
+checkTx(txHash, "execute action")
+```
+
+In the above example, the schema's `"tag"` action is defined as:
+
+```js
+action tag($msg) public {
+    INSERT INTO "tags" (ident, msg) VALUES (@caller, $msg);
+}
+```
+
+We called the `"tag"` action with arguments `[][]any{{"jon was here"}}`. This is
+a `[][]any` to support batched action execution. In this example, we execute the
+action once, using `"jon was here"` as the `$msg` argument.
+
+NOTE: To execute an action with no input arguments, provide `nil`.
+
+### View (read-only) Action Calls
+
+To run a read-only action, which is defined with the `view` modifier, use the
+`CallAction` method.
+
+For example, to call the `get_all` method that returns all records in the `tags` table:
+
+```go
+// Use a read-only view call (no blockchain transaction) to list all entries
+records, err := cl.CallAction(ctx, dbid, "get_all", nil)
+if err != nil {
+	log.Fatal(err)
+}
+```
+
+The `CallAction` method returns the data in the `core/types/client.Records`
+type. See the [godocs](https://pkg.go.dev/github.com/kwilteam/kwil-db/core/types/client)
+for this type to see the methods available for accessing the records.
+
```

Comment:
Do we need an example for authenticating with KGW? Can I just pass a signer to `cl.CallAction`?

Reply:
Kwil has a gateway client that can do that. It is an entirely separate package, which is not what this is documenting/